### DISCUSSION TITLE
Args not being passed to action hosted as container on dockerhub

### DISCUSSION THREAD

**Discussion by brettcannon** [community/community#25084] [Actions]
<p>I <a href="https://github.com/microsoft/pvscbot/tree/gh-action" rel="nofollow noopener">created an action using Python</a> which meant running it as a Docker container and has one required input. Referencing the Dockerfile locally worked fine in <a href="https://github.com/microsoft/pvscbot/blob/master/.github/workflows/dogfood.yml" rel="nofollow noopener">the workflow</a>, although the [16 second build overhead](http://(<a href="https://github.com/microsoft/pvscbot/commit/a1297060c82cc8638c7134745e24961c227126c4/checks?check_suite_id=387301721" class="inline-onebox-loading" rel="nofollow noopener">https://github.com/microsoft/pvscbot/commit/a1297060c82cc8638c7134745e24961c227126c4/checks?check_suite_id=387301721</a>) was unfortunate. I decided to see if pushing the <a href="https://hub.docker.com/repository/docker/brettcannon/pvscbot/general" rel="nofollow noopener">container to dockerhub</a> would help by skipping the build step and thus lower the time it took to execute, but now the action <a href="https://github.com/microsoft/pvscbot/commit/b414352f1eac214ed1d01c6e1d872eaf8dabfdb2/checks?check_suite_id=389244366" rel="nofollow noopener">says there’s no argument being passed in</a> anymore for the input and thus is erroring out. (I even ran it again referencing the action locally and <a href="https://github.com/microsoft/pvscbot/commit/bd3136d5300db05c227e2d68a123edd8760cf156/checks?check_suite_id=389264519" rel="nofollow noopener">it succeeded</a>.)</p>
<p>What am I missing to make calling a container hosted on dockerhub work with an input?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>It’s not supported to set self-defined ‘repo-token’ in ‘jobs.&lt;job_id&gt;.steps.with’ for docker image.</p>
<p>With docker image while not action in ‘uses’, workflow will pull the docker image from public docker hub. According to the syntax <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepswith" rel="nofollow noopener">doc</a>, you can use ‘jobs.&lt;job_id&gt;.steps.with.args’ to pass a command to execute your script, but however cannot set input for the docker image.</p>
<p>Code sample as below:</p>
<pre><code>name: my workflow
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
      - name: Check container
        uses: docker://alpine:3.8
        with:
          args: /bin/sh -c "cat /etc/alpine-release"
</code></pre>

**Comment by brettcannon** [comment]
<p>Thanks! That does indeed solve my immediate problem. It is rather unfortunate, though, that it can’t use the <code>actions.yml</code> file to help make sure people provide the appropriate details. I’ll have to think about whether that loss in the action’s API is worth the 7 second savings in execution time (unless specifying a Docker image action by git tag causes GitHub to cache results?).</p>
